topology/nhlt/intel/dmic/dmic-debug.c

```diff
@@ -0,0 +1,197 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2021 Intel Corporation. All rights reserved.
+//
+// Author: Seppo Ingalsuo <seppo.ingalsuo@linux.intel.com>
+//         Jaska Uimonen <jaska.uimonen@linux.intel.com>
+
+#include <stdint.h>
+#include <stdio.h>
+#include <alsa/input.h>
+#include <alsa/output.h>
+#include <alsa/conf.h>
+#include <alsa/error.h>
+#include "pre-processor.h"
+#include "dmic-debug.h"
+
+#ifdef NHLT_DEBUG
+
+#define MAX_NHLT_BUFFER_SIZE 25000
+
+/* print blob as bytes hex string like: 0x11,0xff,0xff,0xff etc. */
+void dmic_print_bytes_as_hex(uint8_t *src, size_t size)
+{
+	tplg_pp_debug("printing dmic vendor blob as bytes:");
+	char nhlt_string_buffer[MAX_NHLT_BUFFER_SIZE];
+	char *nhlt_p = &nhlt_string_buffer[0];
+	int i;
+
+	if (size >= MAX_NHLT_BUFFER_SIZE) {
+		tplg_pp_debug("printing dmic vendor blob size exceeds MAX_NHLT_BUFFER_SIZE");
+		return;
+	}
+
+	for (i = 0; i < size; i++) {
+		snprintf(nhlt_p, 6, "0x%02x,", *src);
```

Comment:
do we really need the snprintf , why cant we directly do a printf?

Reply:
I mean this is to print the internal byte values as alsa conf hex bytes into a string (which is then added to the conf)?

Reply:
of yes here in the debug, yes sorry will change to printf. This was just mimicking exactly same code as in actual nhlt blob generation in upper levels, will change.